src/main/java/mate/academy/bookshop/controller/BookController.java

```diff
@@ -28,31 +31,37 @@
 public class BookController {
     private final BookService bookService;
 
+    @PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of all available books")
-    public List<BookDto> findAll(Pageable pageable) {
-        return bookService.findAll(pageable);
+    public List<BookDto> findAll(Authentication authentication, Pageable pageable) {
+        User user = (User) authentication.getPrincipal();
+        return bookService.findAll(user.getEmail(), pageable);
```

Comment:
"I'm not sure, but it seems that it's not required in our task. We just need access to all books in the store to choose them from the list."